packages/cli/generators/relation/relationutils.js

```diff
@@ -0,0 +1,158 @@
+'use strict';
+
+const utils = require('../../lib/utils');
+
+exports.relationType = {
+  belongsTo: 'belongsTo',
+  hasMany: 'hasMany',
+  hasOne: 'hasOne',
+};
+
+exports.addFileToProject = function(project, path, modelName) {
+  const fileName = path + '/' + modelName + '.model.ts';
+  return project.addExistingSourceFile(fileName);
+};
+
+exports.getClassObj = function(fileName, modelName) {
+  return fileName.getClassOrThrow(modelName);
+};
+
+exports.getClassesCount = function(fileName) {
+  return fileName.getClasses().length;
+};
+
+exports.isPropertyExist = function(classObj, propertyName) {
+  return classObj
+    .getProperties()
+    .map(x => x.getName())
+    .includes(propertyName);
+};
+
+exports.isRelationExist = function(classObj, propertyName) {
+  if (this.isPropertyExist(classObj, propertyName)) {
+    console.log('property ' + propertyName + ' exsist in the model');
+    throw new Error(' Property exsists');
+  }
+  return;
+};
+
+exports.vlidateType = function(classObj, foriegnKeyName, foriegnKeyType) {
```

Comment:
Typo
```suggestion
exports.vlidateType = function(classObj, foreignKeyName, foriegnKeyType) {
```

Please search for other places containing the same typo and replace `foriegn` to `foreign`.

Reply:
Fixed.